src/main/java/io/cdap/plugin/google/shopping/content/source/ShoppingContentRecordReader.java

```diff
@@ -0,0 +1,60 @@
+package io.cdap.plugin.google.shopping.content.source;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * A {@link RecordReader} to read the {@link ShoppingContentSplit}.
+ */
+public class ShoppingContentRecordReader extends RecordReader<Text, ProductWritable> {
+
+  private List<ProductWritable> productList;
+  private int currentIndex = 0;
+  private ShoppingContentSplit shoppingContentSplit;
+
+  @Override
+  public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptContext) {
+    shoppingContentSplit = (ShoppingContentSplit) inputSplit;
```

Comment:
should create the shopping client here and close it (if needed) in the close() method.

Reply:
done